### ISSUE TITLE
Events are rejected if min_depth is 0 and prev_events refers to missing events

### ISSUE THREAD

**Issue by babolivier** [matrix-org/synapse#6651]
If the server's `min_depth` (i.e. the `min_depth` column from the `room_depth` DB table) for a room is 0 (which means it has seen the first evens for that room), then the server considers it has seen every event in the room and therefore won't ever fetch the `prev_events` for any event sent to it.

This means that, if for some reason it doesn't see an event sent to that room, it's going to reject every event sent to that room that includes the missed event in its `prev_events` because it believes that it has seen every event in the room but doesn't know about the missed events.

**Comment by richvdh** [comment]
> which means it has seen the first evens for that room

I'm pretty sure that the first event in a room is meant to have `depth=1`, so ...?

**Comment by babolivier** [comment]
> > which means it has seen the first evens for that room
> 
> I'm pretty sure that the first event in a room is meant to have `depth=1`, so ...?

Not sure, I've got a fair amount of rooms where `min_depth=0`:

```
synapse=# SELECT COUNT(*) FROM room_depth WHERE min_depth = 0;
 count 
-------
   226
(1 row)
```

**Comment by richvdh** [comment]
it may not matter: just saying I'm not sure it means what you think it means.

**Comment by babolivier** [comment]
> it may not matter: just saying I'm not sure it means what you think it means.

Oh, yes indeed. FWIW I spent some time looking at the code and my server's DB and I've got no idea how `min_depth` could get below 1. Could be a previous version of Synapse, but that happened in a room that was created on Nov 21st, which is not such a long time ago.

Anyway, this has been fixed by https://github.com/matrix-org/synapse/pull/6652 so I'm going to close this.
